akka-cluster/src/main/scala/akka/cluster/ClusterHeartbeat.scala

```diff
@@ -168,6 +177,261 @@ private[cluster] final class ClusterHeartbeatSender extends Actor with ActorLogg
 
 }
 
+/*
+ * INTERNAL API
+ *
+ * This actor is responsible for sending the heartbeat messages to
+ * a few other nodes, which will reply and then this actor updates the
+ * failure detector.
+ * 
+ * TODO: The between teams monitoring 
+ */
+private[cluster] final class CrossDcHeartbeatSender() extends Actor with ActorLogging {
+  import ClusterHeartbeatSender._
+  import CrossDcHeartbeatSender._
+
+  val cluster = Cluster(context.system)
+  val verboseHeartbeat = cluster.settings.Debug.VerboseHeartbeatLogging
+  import cluster.{ selfAddress, selfUniqueAddress, scheduler }
+  import cluster.selfDatacenter
+  import cluster.settings._
+  import context.dispatcher
+
+  // TODO this acts like a vector clock, but only a local one.......
+  var localClock: Int = 0
+  def incLocalClock: Int = {
+    localClock += 1
+    localClock
+  }
+  def incLocalClockJson =
+    s"""{"${selfAddress.host.get}:${selfAddress.port.get}": ${incLocalClock}}"""
+  // TODO END OF this acts like a vector clock, but only a local one.......
```

Comment:
these are needed for using https://bestchai.bitbucket.io/shiviz/ 
![shiviz](https://user-images.githubusercontent.com/120979/27856931-aa8f1c50-61aa-11e7-8c39-c792abd64660.png)

Super useful when working on it; will remove once done debugging